Damage.cs

```diff
@@ -1,65 +1,59 @@
-using System;
+﻿using System;
 using System.Text;
 using System.Diagnostics;
 
 namespace PhysicalDamage.Core
 {
-    abstract class Damage
+    internal abstract class Damage
     {
-        #region DamageType
-        protected EDamageTypes damageType;
-
-        public EDamageTypes DamageType
+        protected Damage(DamageTypes damageType, Target currentTarget)
         {
-            // Once a damage has gained its type (in the constructor), its damage type should remain unchanged
-            get
-            {
-                return damageType;
-            }
+            this.DamageType = damageType;
+            this.CurrentTarget = currentTarget;
         }
-        #endregion
 
-        #region TargetAttribute
-        public struct Target
-        {
-            public float Armor;
+        /// <Summary>
+        ///  A struct containing the data of the target of the damage
+        /// </Summary>
+        public Target CurrentTarget { get; private set; }
 
-            public ERA ERAData;
-        }
+        /// <Summary>
+        ///  The type of the damage, indicated by a enum
+        /// </Summary>
+        public DamageTypes DamageType { get; private set; }
 
-        public struct ERA
+        /// <Summary>
+        ///     Use this method to calculate damage.
+        ///     This method returns a Target struct
+        ///     containing updated values of Health, Armor and ERA stats.
+        /// </Summary>
+        public virtual Target CalculateDamage()
         {
-            public float currentValue;
-
-            public float KEFractionMultiplier;
-
-            public float HEATFractionMultiplier;
+            // Override this function to specify damage algorithm
+            return this.CurrentTarget; // No damage dealt thus return the original state of the target
         }
-        #endregion
 
-        public Damage(EDamageTypes dt)
+        public struct Era
```

Comment:
Now that you've started getting the hang of things. Consider another option instead of three different structs. Maybe a `class`. *wink* *wink*